modules/videoio/cmake/init.cmake

```diff
@@ -1,31 +1,19 @@
-macro(add_backend backend_id cond_var)
-  if(${cond_var})
-    include("${CMAKE_CURRENT_LIST_DIR}/detect_${backend_id}.cmake")
-  endif()
-endmacro()
+include(FindPkgConfig)
 
-function(ocv_add_external_target name inc link def)
-  if(BUILD_SHARED_LIBS)
-    set(imp IMPORTED)
-  endif()
-  add_library(ocv.3rdparty.${name} INTERFACE ${imp})
-  set_target_properties(ocv.3rdparty.${name} PROPERTIES
-    INTERFACE_INCLUDE_DIRECTORIES "${inc}"
-    INTERFACE_SYSTEM_INCLUDE_DIRECTORIES "${inc}"
-    INTERFACE_COMPILE_DEFINITIONS "${def}")
-  # When cmake version is greater than or equal to 3.11, INTERFACE_LINK_LIBRARIES no longer applies to interface library
-  # See https://github.com/opencv/opencv/pull/18658
-  if (CMAKE_VERSION VERSION_LESS 3.11)
-    set_target_properties(ocv.3rdparty.${name} PROPERTIES
-      INTERFACE_LINK_LIBRARIES "${link}")
-  else()
-    target_link_libraries(ocv.3rdparty.${name} INTERFACE ${link})
-  endif()
-  #
-  if(NOT BUILD_SHARED_LIBS)
-    install(TARGETS ocv.3rdparty.${name} EXPORT OpenCVModules)
-  endif()
-endfunction()
+# FIXIT: stop using PARENT_SCOPE in dependencies
+if(PROJECT_NAME STREQUAL "OpenCV")
+  macro(add_backend backend_id cond_var)
+    if(${cond_var})
+      include("${CMAKE_CURRENT_LIST_DIR}/detect_${backend_id}.cmake")
+    endif()
+  endmacro()
+else()
+  function(add_backend backend_id cond_var)
+    if(${cond_var})
+      include("${CMAKE_CURRENT_LIST_DIR}/detect_${backend_id}.cmake")
+    endif()
+  endfunction()
+endif()
```

Comment:
There is inconsistency in scope of dependencies.

Reply:
Perhaps we could switch to target checks instead of `HAVE_` variables. Additional information can be stored in custom target properties.